File: src/contests/cg/puzzle/telephone-numbers.ts
// @ts-nocheck
// 🎮 CodinGame Puzzle - telephone-numbers
// https://www.codingame.com/training/medium/telephone-numbers

const n: number = parseInt(readline());
const root: Record<string, any> = {};
let cableUnits: number = 0;

for (let i = 0; i < n; i++) {
  const number: string = readline().trim();
  let node: Record<string, any> = root;
  for (const digit of number) {
    if (!(digit in node)) {
      node[digit] = {};
      cableUnits++;
    }
    node = node[digit];
  }
}

console.log(cableUnits);

File: src/contests/cg/puzzle/python3/telephone-numbers.py
# 🎮 CodinGame Puzzle - telephone-numbers
# https://www.codingame.com/training/medium/telephone-numbers

n = int(input())
root = {}
cable_units = 0
for _ in range(n):
    number = input().strip()
    node = root
    for digit in number:
        if digit not in node:
            node[digit] = {}
            cable_units += 1
        node = node[digit]
print(cable_units)
